src/tools/generators.ts

```diff
@@ -245,3 +259,279 @@ export function installGenerators(generators: string[]): string[] {
+            directories: true,
+            files: false,
+            matching: "AppIcon.appiconset",
+          })?.[0]) ||
+        "ios/**/Images.xcassets/AppIcon.appiconset",
+    }[o]
+    const outputDirPath = path(cwd, relativeOutputDirPath)
+
+    // if not, skip...
+    if (exists(outputDirPath) !== "dir") {
+      warning(
+        `⚠️  No output directory found for "${optionProjectName}" at "${outputDirPath}". Skipping...`,
+      )
+      continue
+    }
+
+    heading(`Generating ${optionProjectName} app icons...`)
+
+    const icons = APP_ICON_RULES.icons.filter((i) => i.platform === o)
+
+    // prepare each icon for generation sequentially
+    for (const i of icons) {
+      const inputFilePath = path(templatesDir(), "app-icon", i.inputFile)
+
+      // get the input file for sharp and do some initial transforms when necessary (e.g. radius and padding)
+      const inputFile = await (async function () {
+        if (!i.transform) return inputFilePath
+
+        try {
+          const { size, radius, padding } = i.transform
+          const cutoutMask = Buffer.from(
+            `<svg><rect x="0" y="0" width="${size}" height="${size}" rx="${radius}" ry="${radius}"/></svg>`,
+          )
+          return await sharp(inputFilePath)
+            .resize(size, size, { fit: "fill" })
+            .composite([{ input: cutoutMask, blend: "dest-in" }])
+            .extend({
+              top: padding,
+              bottom: padding,
+              left: padding,
+              right: padding,
+              background: { r: 0, g: 0, b: 0, alpha: 0 },
+            })
+            .toBuffer()
+        } catch (error) {}
+      })()
+
+      const rules = APP_ICON_RULES.rules.filter((i) => i.platform === o)
+
+      // actually resize the input files and save to output dir sequentially
+      for (const r of rules) {
+        // construct the output file name
+        const outputFileName = {
+          expo: i.name,
+          android: i.name.replace("{dpi}", r.dpi),
+          ios: i.name
+            .replace("{size}", r.size[i.type])
+            .replace("{idiom}", r.idiom)
+            .replace("{scale}", r.scale > 1 ? `@${r.scale}x` : ""),
+        }[o]
+
+        if (!inputFile) {
+          warning(`⚠️  ${outputFileName}: transform failed, please file an issue on GitHub`)
+          continue
+        }
+
+        const outputFilePath = path(outputDirPath, outputFileName)
+        const outputSize = r.size[i.type] * (r.scale || 1)
+
+        // finally, resize and save
+        try {
+          await sharp(inputFile)
+            .resize(outputSize, outputSize, { fit: "fill" })
+            .toFile(outputFilePath)
+
+          direction(`✅ ${outputFileName}`)
+        } catch (error) {
+          warning(`⚠️  ${outputFileName}: saving failed, check if the directory exists`)
+        }
+      }
+    }
+
+    const boilerplateDirPath = path(igniteCliRootDir(), "boilerplate")
+
+    const postGenerationFn = {
```

Comment:
after the generation is done, we make sure the each platform is actually referencing the new files. in most cases, for people using ignite version that this will release in, that will not be necessary since the defaults ship in the boilerplate. However, if people already modified the necessary files, we make sure to make the needed changes.

Reply:
Yeah these types of checks are really nice for a good DX. :100: